src/component.ts

```diff
@@ -1,16 +1,56 @@
-import { BaseSDK } from "./base";
-import { LISTENER_CMDS } from "./constants";
+import { BaseSDK, EventBase } from "./base";
+import { LISTENER_CMDS, EVENT_TYPES } from "./constants";
+
+import { globalInstances } from "./utils";
 
 import { ComponentProps } from "./types/internal";
 
 export class Component extends BaseSDK {
 	_id: string;
 	type: string;
+
+	#registerComponentAPIs(componentAPIs) {
+		componentAPIs?.forEach((Api) => {
+			this[Api.name] = (...args) => {
+				if (Api.type === "method") {
+					return this._postMessageAsync(`COMPONENT_${Api.name}`, {
+						id: this._id,
+						parameters: args
+					});
+				} else if (Api.type === "event") {
+					this._postMessage(
+						LISTENER_CMDS.COMPONENT_ADD_EVENT_LISTENER,
+						{
+							id: this._id,
+							eventName: Api.name,
+							eventConfig: Api.eventConfig
+						}
+					);
+					this._addEventListener(Api.name, args[0]);
+				}
+			};
+		});
```

Comment:
we should auto listen for events, unless user has explicitly adding listeners in their code.